tests/contrib/aiopg/test_aiopg.py

```diff
@@ -0,0 +1,209 @@
+import sys
+from textwrap import dedent
+import time
+
+# 3p
+import aiopg
+from psycopg2 import extras
+from nose.tools import eq_
+
+# project
+from ddtrace.contrib.aiopg.patch import patch, unpatch
+from ddtrace import Pin
+
+# testing
+from tests.contrib.config import POSTGRES_CONFIG
+from tests.test_tracer import get_dummy_tracer
+from tests.contrib.asyncio.utils import AsyncioTestCase, mark_asyncio
+
+
+TEST_PORT = str(POSTGRES_CONFIG['port'])
+PY35 = sys.version_info >= (3, 5)
+
+
+class TestPsycopgPatch(AsyncioTestCase):
+    # default service
+    TEST_SERVICE = 'postgres'
+
+    def setUp(self):
+        super().setUp()
+        self._conn = None
+        patch()
+
+    def tearDown(self):
+        super().tearDown()
+        if self._conn and not self._conn.closed:
+            self._conn.close()
+
+        unpatch()
+
+    @asyncio.coroutine
+    def _get_conn_and_tracer(self):
+        conn = self._conn = yield from aiopg.connect(**POSTGRES_CONFIG)
+        Pin.get_from(conn).clone(tracer=self.tracer).onto(conn)
+
+        return conn, self.tracer
+
+    @asyncio.coroutine
+    def assert_conn_is_traced(self, tracer, db, service):
+
+        # ensure the trace aiopg client doesn't add non-standard
+        # methods
+        try:
+            yield from db.execute("select 'foobar'")
+        except AttributeError:
+            pass
+
+        writer = tracer.writer
+        # Ensure we can run a query and it's correctly traced
+        q = "select 'foobarblah'"
+        start = time.time()
+        cursor = yield from db.cursor()
+        yield from cursor.execute(q)
+        rows = yield from cursor.fetchall()
+        end = time.time()
+        eq_(rows, [('foobarblah',)])
+        assert rows
+        spans = writer.pop()
+        assert spans
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.name, "postgres.query")
+        eq_(span.resource, q)
+        eq_(span.service, service)
+        eq_(span.meta["sql.query"], q)
+        eq_(span.error, 0)
+        eq_(span.span_type, "sql")
+        assert start <= span.start <= end
+        assert span.duration <= end - start
+
+        # run a query with an error and ensure all is well
+        q = "select * from some_non_existant_table"
+        cur = yield from db.cursor()
+        try:
+            yield from cur.execute(q)
+        except Exception:
+            pass
+        else:
+            assert 0, "should have an error"
+        spans = writer.pop()
+        assert spans, spans
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.name, "postgres.query")
+        eq_(span.resource, q)
+        eq_(span.service, service)
+        eq_(span.meta["sql.query"], q)
+        eq_(span.error, 1)
+        eq_(span.meta["out.host"], "localhost")
+        eq_(span.meta["out.port"], TEST_PORT)
+        eq_(span.span_type, "sql")
+
+    if PY35:
+        # We have to exec this due to syntax errors on earlier versions.
+        exec(dedent("""
```

Comment:
I think we should not do that way. Here is better to put this test in a separate file and exclude it when python 3.4 is used.